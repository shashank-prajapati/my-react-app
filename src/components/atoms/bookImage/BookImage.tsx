import React from "react";

type Props = {
  source: string;
  width?: string | number | undefined
};

function BookImage(props: Props) {
  const { source,width } = props;
  return (
    <img
      style={{ display: "block" }}
      src={source}
      width={width}
      alt="Book Cover"
    ></img>
  );
}

export default BookImage;
