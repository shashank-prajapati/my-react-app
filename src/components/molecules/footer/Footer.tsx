import { Paper, Typography } from '@mui/material';
import React from 'react';
import BlinkistHeaderLogo from '../../atoms/blinkistHeaderLogo/BlinkistHeaderLogo';
import Grid from '@mui/material/Grid';
import FooterList from '../footerList/FooterList';


type Props = {};

const list1=['Book lists','What is Nonfiction?','What to read next?','Benefits of reading'];
const list2 =['Pricing','Blinkist business','Gift cards','Blinkist magaine','Contact & help'];
const list3 = ['About','Careers','partners','Code of Conduct'];

function Footer({}: Props) {
  return <Paper sx={{
      px:'200px',
      bgcolor:'footerBgColor.main',
      py:'38px',
      mt:'100px'
  }}>
      <Grid container >
          <Grid item sm={6}>
            <BlinkistHeaderLogo width='100px' height='24px'/>
            <Typography sx={{mt:8}} variant='subtitle1' color='secondary'>Big ideas in small packages Start learning now</Typography>
          </Grid>
          <Grid item sm={2}>
            <FooterList heading='Editorial' list={list1}/>
          </Grid>
          <Grid item sm={2}>
          <FooterList heading='Useful links' list={list2}/>
          </Grid>
          <Grid item sm={2}>
          <FooterList heading='Company' list={list3}/>
          </Grid>
      </Grid>
      <Typography variant='caption1' color='extraTextColor.main'>Â© Blinkist 2021 Sitemap   |   Imprint   |   Terms of Service   |   Privacy Policies</Typography>
  </Paper>;
}

export default Footer;
