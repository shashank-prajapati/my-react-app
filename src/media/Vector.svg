<svg width="16" height="23" viewBox="0 0 16 23" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M11.502 19C10.8404 20.5823 9.58232 21.8404 8 22.502C6.41768 21.8404 5.15956 20.5823 4.498 19H6.758C7.084 19.489 7.505 19.912 8 20.243C8.495 19.913 8.916 19.489 9.243 19H11.502ZM14 13.805L16 16.073V18H0V16.073L2 13.805V7.99996C2 4.51696 4.504 1.55296 8 0.454956C11.496 1.55296 14 4.51696 14 7.99996V13.805ZM13.27 16L12 14.56V7.99996C12 5.68196 10.43 3.56996 8 2.57996C5.57 3.56996 4 5.68096 4 7.99996V14.56L2.73 16H13.27ZM8 9.99996C7.46957 9.99996 6.96086 9.78924 6.58579 9.41417C6.21071 9.0391 6 8.53039 6 7.99996C6 7.46952 6.21071 6.96082 6.58579 6.58574C6.96086 6.21067 7.46957 5.99996 8 5.99996C8.53043 5.99996 9.03914 6.21067 9.41421 6.58574C9.78929 6.96082 10 7.46952 10 7.99996C10 8.53039 9.78929 9.0391 9.41421 9.41417C9.03914 9.78924 8.53043 9.99996 8 9.99996Z"/>
</svg>
