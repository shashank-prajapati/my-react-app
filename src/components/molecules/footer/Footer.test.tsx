import * as React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import Footer from "./Footer";
import { Typography } from "@mui/material";
import BookImage from "../../../media/coverImage.png";
import { ReactComponent as EntrepreneurshipIcon } from "../../../media/Vector.svg";

describe("Testing Footer", () => {
  it("should match the Footer snapshot", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
