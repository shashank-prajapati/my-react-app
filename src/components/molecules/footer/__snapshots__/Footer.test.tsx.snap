// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Footer should match the Footer snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1hvgpz9-MuiPaper-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-6 css-6dzwqk-MuiGrid-root"
      >
        <img
          alt="Book"
          height="24px"
          src="blinkistLogo.png"
          style="display: block;"
          width="100px"
        />
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 css-1e76e2v-MuiTypography-root"
        >
          Big ideas in small packages Start learning now
        </h6>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2 css-1h3qeln-MuiGrid-root"
      >
        <div>
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1cka7m9-MuiTypography-root"
              >
                Editorial
              </p>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1pjftnz-MuiTypography-root"
              >
                Book lists
              </p>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1pjftnz-MuiTypography-root"
              >
                What is Nonfiction?
              </p>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1pjftnz-MuiTypography-root"
              >
                What to read next?
              </p>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1pjftnz-MuiTypography-root"
              >
                Benefits of reading
              </p>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2 css-1h3qeln-MuiGrid-root"
      >
        <div>
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1cka7m9-MuiTypography-root"
              >
                Useful links
              </p>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1pjftnz-MuiTypography-root"
              >
                Pricing
              </p>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1pjftnz-MuiTypography-root"
              >
                Blinkist business
              </p>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1pjftnz-MuiTypography-root"
              >
                Gift cards
              </p>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1pjftnz-MuiTypography-root"
              >
                Blinkist magaine
              </p>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1pjftnz-MuiTypography-root"
              >
                Contact & help
              </p>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2 css-1h3qeln-MuiGrid-root"
      >
        <div>
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1cka7m9-MuiTypography-root"
              >
                Company
              </p>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1pjftnz-MuiTypography-root"
              >
                About
              </p>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1pjftnz-MuiTypography-root"
              >
                Careers
              </p>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1pjftnz-MuiTypography-root"
              >
                partners
              </p>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1pjftnz-MuiTypography-root"
              >
                Code of Conduct
              </p>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <span
      class="MuiTypography-root MuiTypography-caption1 css-1svm2tk-MuiTypography-root"
    >
      Â© Blinkist 2021 Sitemap | Imprint | Terms of Service | Privacy Policies
    </span>
  </div>
</div>
`;
